### PR TITLE
BC-239 - normalize students birth date

### PR DESCRIPTION
**Pull request by uidp**
# Description

BC-239 - normalize students birth date

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: helpers/date.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const moment = require('moment');
```
Comment (tofixx) [inline helpers/date.js:1]: please consider,  using the datetime util for language dependant date formatting and send an isostring of the date to the server instead. https://github.com/hpi-schul-cloud/schulcloud-client/blob/main/static/scripts/datetime/datetime.js

**Review by tofixx — CHANGES_REQUESTED** [review]

**Review by CeEv — APPROVED** [review]

File: helpers/date.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const DATE_FORMATS = {
+	de: 'DD.MM.YYYY',
+	en: 'MM/DD/YYYY',
+	es: 'MM/DD/YYYY',
+};
+
+const normalizeDate = (srcDateString, lang) => {
+	const srcFormat = DATE_FORMATS[lang];
+	// we do not allow undefined as input => set to invalid empty string
+	const date = moment(srcDateString || '', srcFormat);
```
Comment (tofixx) [inline helpers/date.js:12]: srcDateString should be an non whitespace and non empty string of length 10

File: helpers/date.js (hunk 1 above)
Comment (uidp) [reply helpers/date.js:1]: we changed to ISO string

**Review by CeEv — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +48,18 @@ Allowed Types of change: `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `
 
 - BC-137 - fix bug with display of Blue Bar
 
+## [26.9.3] - 2021-09-10
+
+### Fixed
+
+- BC-239 - normalize students birth date
```
Comment (tofixx) [inline CHANGELOG.md:55]: are the changes in this file correct?!

File: test/mocha/helpers/date.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+		expect(() => normalizeDate('20.01.2004', 'fr')).to.throw();
+	});
+
+	it('should throw if no language is provided', () => {
+		expect(() => normalizeDate('20.01.2004')).to.throw();
+		expect(() => normalizeDate('20.01.2004', '')).to.throw();
+	});
+
+	it('should throw if the date string is invalid', () => {
+		expect(() => normalizeDate('20/01.2004')).to.throw();
```
Comment (tofixx) [inline test/mocha/helpers/date.test.js:21]: should throw when date/format combination does not match?

**Review by tofixx — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (CeEv) [reply CHANGELOG.md:55]: yes ..the basic hotfix is go out as 9.3. but is must also merge to the already exist release 26.10.x and by merge this i make a misstake and we must solve this misstake 26.10.2...not nice but seldom.

**Pull request merged**